test/test_file.cpp

```diff
@@ -408,3 +399,120 @@ TORRENT_TEST(stat_file)
 	TEST_CHECK(ec);
 	TEST_EQUAL(ec, boost::system::errc::no_such_file_or_directory);
 }
+
+// specificaly UNC tests
+#if TORRENT_USE_UNC_PATHS
+
+int maximum_component_length { 0 };
+bool support_hard_links { false };
+
+bool fill_current_directory_caps()
+{
+#ifdef TORRENT_WINDOWS
+	error_code ec;
+	DWORD dw_maximum_component_length;
+	DWORD dw_file_system_flags;
+	if (!GetVolumeInformationA(nullptr, nullptr, 0, nullptr, &dw_maximum_component_length, &dw_file_system_flags, nullptr, 0))
+	{
+		ec.assign(GetLastError(), system_category());
+		std::printf("GetVolumeInformation: [%s] %s\n"
+			, ec.category().name(), ec.message().c_str());
+		return false;
+	}
+	maximum_component_length = int(dw_maximum_component_length);
+	support_hard_links = ((dw_file_system_flags & FILE_SUPPORTS_HARD_LINKS) != 0);
+	return maximum_component_length > 0;
+#else
+	return false;
```

Comment:
it's not obvious to me why this returns false on non-windows. that excludes a bunch of tests from running.

Is this ever expected to fail on windows? I'm kind of thinking that it might be reasonable to throw on failures.